tests/test_register_pending.py

```
@@ -0,0 +1,177 @@
+    # lifespan retry 루프 시뮬레이션: asyncio.sleep 패치 (대기 없이 즉시 진행)
+    import asyncio
+
+    monkeypatch.setattr(
+        asyncio, "sleep", lambda *a, **kw: asyncio.coroutine(lambda: None)()
+    )
+
+    # register_to_backend_pending 자체는 3회 시도 시 마지막에 HTTPStatusError raise
+    # lifespan 분기 2는 이 예외를 잡아 soft-fail 처리 — 단위 테스트에서는 raise 확인
+    attempt_count = 0
+    pending_registered = False
+    for attempt in range(3):
```

Ruff (0.15.12)

[warning] Loop control variable `attempt` not used within loop body

Rename unused `attempt` to `_attempt`

(B007)